### ISSUE TITLE
Fable testing-holiday app (part 1)

### ISSUE THREAD

**Issue by Lu-stack-art** [cds-snc/design-gc-conception#1237]
Screen Reader, VoiceOver with Safari, mobile- iOS
1. Items on page is not visible to VoiceOver
2. Content on page does not seem to be interacting with VoiceOver
3. User's experience indicates that the site is not working, as link to a new page does not display any content, at some point in their journey
4. User is unable to distinguish between link or button "controls need to be labelled better, so that VoiceOver can identify the correct elements (ex, do not use "tab" in taskflow)
5. User suggests to have elements coded, when VoiceOver is going onto a new page, so that VoiceOver can narrate if link takes user to a new page, or opens a new tab
6. When on the About this app page, the VoiceOver seems to only narrate the Code, Design, and Guidance badges, just as text, versus opening the links associated with these tabs/badges onto new pages

https://app.makeitfable.com/requests/CDS-CGDS-27
https://cds-snc.dovetail.com/data/Compatibility-testing-with-fable-holiday-app-alt-nav-screen-reader-and-screen-magnification-4p8N3ZcjoBzscrqL1BZwZl

**Comment by Lu-stack-art** [comment]
Alt Nav user, using Apple’s voice control, on a desktop, on a Mac, using the Edge browser

- Language toggle changed colours to blue when activated, but page did not change to FR webpage

User tried with numbers option, and with grid option on the AT, and no changes have been made to new language page
Speech recognition software appears to have 3 different labels with the same button

What could’ve just been a one command click, ends up being a few clicks
When prompted to show the grid to get to the next step (when will this holiday occur?) the grid does not easily read the number (move up and down command sometimes changes the response from step above)

The radio buttons  are activating many different responses (grid, numbers, and voice control) which interrupts the user’s flow when trying to fill in the form fields
User would like feedback/confirmation that file has been successfully uploaded (as it’s not showing up on this AT/page)
https://app.makeitfable.com/requests/CDS-CGDS-38

**Comment by Lu-stack-art** [comment]
Alternative navigation, on a desktop, on the Edge browser, Windows, using Dragon Naturally Speaking AT

- When user prompts software to submit a holiday, it is not responsive, but instead used a command called ‘show links’, which will look for buttons, links or anything interactable, and highlight them by using numbers on the page.
- user tried directly interacting with the link using command voice but was unresponsive
- voice command is not being detected on this software…meanwhile user was able to type text in textarea component, by dictating the content/spelling out the words
- When user uses ‘show links’ command to choose the select component, (for the month), a number does not pop up next to that drop down, but instead user commands ‘show grid’ which narrows boxes more closely to the area of the page user is trying to select (user could not choose the select component) (select month by name, or by show links-not able to number that list box)…had to use mouse grid which is slow and tedious method
- best command that works on this software for this user is the ‘show links’ command
- some form fields do not acknowledge when dragon enters text…the form field thinks it is empty when it isn’t
- When filling in the email form field/textarea, voice command again was not responsive through Dragon, so user had to spell the text out which is slow and tedious (unable to dictate the email)

- feedback-make sure the text fields allow for spelling dialogue, and dragon be used, so that text can be fully dictated and corrected

- Text field reseting itself and not holding info, is another issue
- https://app.makeitfable.com/requests/CDS-CGDS-42

**Comment by Lu-stack-art** [comment]
2-Alternative Navigation Dragon NaturallySpeaking with Firefox on Windows
- Link or button can't be activated by voice commands (Voice command was not being responsive to the links and buttons)

- Alternative Navigation Voice Control with Safari on Mac
- Notification/confirmation for successful uploading of file is necessary here…tester cannot confirm if task is complete
- voice command is not responsive…the voice control is reading multiple labels on the text box, and not distinguishing labels displaying on buttons and links
- https://app.makeitfable.com/requests/CDS-CGDS-39

**Comment by Lu-stack-art** [comment]
Alternative Navigation Switch system with Chrome on Android

- the links on the side navigation is difficult to find, on a mobile device, the drop down is not as easily accessible
- uploading a file was difficult as it wasn’t opening a pdf doc on the mobile device/or perhaps the AT does not accept the PDF file

- Alternative Navigation Voice Access (Android) with Chrome on Android
- text is too small on mobile device, tester has an easier time locating the links on a desktop versus mobile device
- https://app.makeitfable.com/requests/CDS-CGDS-33

**Comment by adriannelee** [comment]
> 2-Alternative Navigation Dragon NaturallySpeaking with Firefox on Windows
> 
> Link or button can't be activated by voice commands (Voice command was not being responsive to the links and buttons)
> 
> Alternative Navigation Voice Control with Safari on Mac
> 
> Notification/confirmation for successful uploading of file is necessary here…tester cannot confirm if task is complete
> 
> voice command is not responsive…the voice control is reading multiple labels on the text box, and not distinguishing labels displaying on buttons and links
> 
> https://app.makeitfable.com/requests/CDS-CGDS-39

Added to [#2092](https://app.zenhub.com/workspaces/gc-design-system-6100624a19f4cf000e46e458/issues/gh/cds-snc/design-gc-conception/2092)

**Comment by adriannelee** [comment]
> Alternative Navigation Switch system with Chrome on Android
> 
> the links on the side navigation is difficult to find, on a mobile device, the drop down is not as easily accessible
> 

Added to [#1133](https://app.zenhub.com/workspaces/gc-design-system-6100624a19f4cf000e46e458/issues/gh/cds-snc/design-gc-conception/1842). 

> uploading a file was difficult as it wasn’t opening a pdf doc on the mobile device/or perhaps the AT does not accept the PDF file
> 

It's unclear which of the 5 testers gave this feedback, on which technology, and what it's about exactly.

> Alternative Navigation Voice Access (Android) with Chrome on Android
> 
> text is too small on mobile device, tester has an easier time locating the links on a desktop versus mobile device

This feedback was not found on this request.

> 
> https://app.makeitfable.com/requests/CDS-CGDS-33
